imod/prepare/spatial.py

```diff
@@ -292,6 +294,8 @@ def _handle_dtype(dtype, nodata):
         "int32": (-2147483648, 2147483647),
         "float32": (-3.4028235e38, 3.4028235e38),
         "float64": (-1.7976931348623157e308, 1.7976931348623157e308),
+        "uint64": (0, 18446744073709551615),
+        "int64": (-9223372036854775808, 9223372036854775807),
```

Comment:
Nice that this got an update, I infer from rasterio source code that this has been introduced in GDAL 3.5; GDAL is quite notorious for contributing to dependency hell in Python projects in the past, though it has recently gotten a lot better. 

I think to be on the safe side, we should either add a GDAL version check here (like rasterio does) or restrict ourselves to GDAL 3.5 or higher in the pixi or pyproject.toml. 

Versionwise, what is common now:
GDAL 3.5 was released in may 2022. I see rasterio requires GDAL 3.3 or higher in their setup.py. So I think it is fine to enforce GDAL 3.5 or higher.

Reply:
Restricted the version of GDAL to a minimum of 3.5